mito-ai/src/Extensions/AiChat/ChatMessage/ChatMessage.tsx

```diff
@@ -36,20 +37,72 @@ const ChatMessage: React.FC<IChatMessageProps> = ({
     operatingSystem,
     setDisplayCodeDiff,
     acceptAICode,
-    rejectAICode
+    rejectAICode,
+    onUpdateMessage
 }): JSX.Element | null => {
+    const [isEditing, setIsEditing] = useState(false);
```

Comment:
Maybe to consolidate state we can just have `editedContent` that is either a string or undefined. If it's undefined we are not in editing mode. If it's a string we are in editing mode?

Then when, `handleEditClick` is called, we can do this 

```
(message.content as string).replace(/```[\s\S]*?```/g, '').trim()
```

And when `handleCancel` is pressed, all we need to do is set `editedContent` to `undefined` instead of recalculating the original editing message.

Reply:
I'm going to mark this as resolved for now. After moving `ChatInput` into it's own component, `editedContent` and `isEditing` are no longer in the same file. This removes the option to consolidate.

Feel free to unresolved if you feel otherwise.